components/AddOrEdit.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import EditModal from "./EditModal";
+import CreatePost from "./CreatePost";
+import Product from "@/Types/Product";
+
+// { pageIdentifier }: { pageIdentifier: string }
+const AddOrEdit = ({
+  pageIdentifier,
+  product,
+  editModal,
+  handleClose,
+  categories,
+}: {
+  product: Product | null;
+  editModal: boolean;
+  handleClose: () => void;
+  categories: string[];
+  pageIdentifier: string;
+}) => {
+  return (
```

Comment:
This could also be done like:
`if (pageIdentifier === 'editModal') return <EditModal ... />`
`return <CreatePost />``